### ISSUE TITLE
Slow loading of table with large geometry objects

### ISSUE THREAD

**Issue by jczaplew** [jakob/Postico#212]
I have a table with 37 rows, each with a fairly large geometry. If I `SELECT * FROM _table_` it takes about 15s, but if I simply load the table from the Postico browser it takes more than 30s.

My guess is that Postico is selecting the geometry `ST_AsEWKT()`, which is slowing the query down. If I `SELECT *, ST_AsEWKT(geom) FROM _table_` the query takes approximately as long as Postico takes to load it.

I can see a few different solutions to this problem:

1. Simply select the binary geometry
2. If the geometry is requested and not transformed (either as a part of `*` or `SELECT geom`), return some sort of placeholder. Perhaps something like `<Geometry>`
3. Similar to number 2, but return the type of geometry with `SELECT GeometryType(geom)`

Thank you for creating such a great product!

**Comment by jakob** [comment]
Yes, Postico uses `ST_AsEWKT()`, since selecting the binary geometry is not very useful in many cases (discussed in issue #92).

The problem with your suggestions is that it would become impossible to view / edit geometries.

Anyway, I assume that the problem isn't that the query itself is slow, but transferring the data takes a lot of time? Performance issues with large rows are not unique to geometry objects, Postico is also slow when rows contain large `bytea` columns.

Maybe a possible solution would be to allow hiding certain columns? Then you could just hide columns that contain lots of data, if you aren't actually interested in it.

**Comment by jczaplew** [comment]
That makes sense, and now I understand the issue. The problem is definitely in the transfer of the data.

This really gets into how individuals manage their data. For example, I don't think I have ever manually edited a geometry nor would I ever want to, but for some people that might be an important part of their workflow. However, I can't imagine anyone manually editing a geometry more complicated than a point or perhaps a polygon with a few vertices. It would be incredibly cumbersome and have the potential for introducing a lot of error.

If you are interested in a geometry-specific solution, something like the following might work:

`
SELECT 
   CASE WHEN octet_length(geom::bytea) > 1000 THEN GeometryType(geom)
   ELSE ST_AsEWKT(geom) 
   END
`

That way smaller geometries that one might want to edit are available, whereas larger ones are hidden. 

Of course, another option would be to add a setting for such a thing, but Postico currently doesn't have a concept of table-specific settings and I think that would be a difficult UI problem.

If you are interested in a solution more general to data types other than geometries, the addition of a setting to hide objects larger than some length would work.

The bigger question: is this an issue that other users have that warrants some sort of enhancement? If so, is a geometry-specific or more general solution a better approach?

**Comment by jczaplew** [comment]
A fix, even if it is simply the ability to disable displaying geometry columns, would be greatly appreciated. I have many tables that if I accidentally click on them in Postico will freeze the application for 5+ minutes while it attempts to load the geometry values.

**Comment by jakob** [comment]
You can press Cmd-. to cancel the current query.

**Comment by jczaplew** [comment]
Awesome! That helps a lot!
